hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -437,12 +438,30 @@ public OzoneVolume getVolumeDetails(String volumeName)
   }
 
   @Override
-  public OzoneVolume getS3VolumeDetails() throws IOException {
-    OmVolumeArgs volume = ozoneManagerClient.getS3Volume();
+  public OzoneVolume getS3Volume() throws IOException {
```

Comment:
I think this method is better fit for `ObjectStore` than `RpcClient`. Putting it here seems to imply it corresponds to an RPC endpoint, when really it is a helper method that delegates to a differently named RPC call. Moving it to `ObjectStore` would make it function similarly to `ObjectStore#creates3Bucket`. There's no `createS3Bucket` RPC, but the method uses other RPCs under the hood. It is also not really part of the 'protocol', so moving it to `Objectstore` would remove it from the `ClientProtocol` interface and clear up some confusion.

Reply:
I was thinking the same. The only issue was that `buildOzoneVolume()` requires a `conf` that is not exposed to `ObjectStore`. I guess I will just add a `RpcClient#getConf` and use that.

Reply:
Done!